packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -289,10 +289,14 @@ export function createVM<HostNode, HostElement>(
 
     let shadowMode;
     if (renderer.syntheticShadow) {
```

Comment:
Looking at this PR, I would recommend changing the name of the `renderer.syntheticShadow` to something like `renderer.isSyntheticShadowDefined` or `renderer.isSyntheticShadowLoaded`. And move the `isNativeShadowRootDefined` to the renderer instead of looking it up from the `globalThis`.

Reply:
> And move the isNativeShadowRootDefined to the renderer instead of looking it up from the globalThis.

I would still need to capture this information before the ShadowRoot polyfill and then somehow communicate it to the engine. Open to suggestions!

Reply:
Here is a proposal https://github.com/salesforce/lwc/pull/2485. Let me know what you think of this approach @ekashida.